test/contract.test.js

```diff
@@ -56,6 +56,32 @@ const contract = new Contract(account, null, {
     });
 });
 
+describe('viewMethod', () => {
+    test('passes options through to account viewFunction', async () => {
+        function customParser () {}
+        const stubbedReturnValue = await contract.viewMethod({}, { parse: customParser });
+        expect(stubbedReturnValue.options.parse).toBe(customParser);
+    });
+
+    test('throws PositionalArgsError if second argument is not an object', () => {
+        return Promise.all([
+            1,
+            'lol',
+            [],
+            new Date(),
+            null,
+            new Set(),
+        ].map(async badArgs => {
+            try {
+                await contract.viewMethod({ a: 1 }, badArgs);
+                throw new Error(`Calling \`contract.viewMethod({ a: 1 }, ${badArgs})\` worked. It shouldn't have worked.`);
```

Comment:
nit: more idiomatic would be to use `fail`

Reply:
Not sure I understand. We want to await the call, and if it succeeds, we throw an error. I spent a significant amount of time trying to find a more idiomatic-to-Jest way of doing this in the similar test above, but ultimately landed back on this approach, which I stole from elsewhere in this codebase.